src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -3620,16 +3654,19 @@ private int indexOf(Rope sourceRope, Rope otherRope, int fromIndex) {
             return -1;
         }
 
-        protected static boolean isSingleByteSearch(DynamicObject source, DynamicObject pattern) {
-            assert RubyGuards.isRubyString(source);
-            assert RubyGuards.isRubyString(pattern);
+        protected static boolean isSingleBytePattern(DynamicObject pattern) {
+            if (RubyGuards.isRubyString(pattern)) {
```

Comment:
Is there a case where `pattern` is not a String?

Reply:
It's a non-standard method. As far as I can tell, it's only called with String and only supported with String. But someone could call it with any object they want. Then you're left with a philosophical debate over whether to try to coerce or raise an exception on bad input.

Reply:
Please let it raise an exception, defensive code handling non-existing cases makes things more complex.
It's an internal primitive and should only be called with appropriate arguments.

Reply:
It is raising an exception for an unsupported specialization. I take it you mean an exception for working with a bad type?

Reply:
I mean no special code handling it and let it raise whatever the safeguards we have in place do.